### PR TITLE
Fix for scheduler bugs

### PR DESCRIPTION
**Pull request by xiulipan**
Now both ipc process task and pipeline copy task are using scheduler. Need to care about the run order of the task. Fix some bugs in scheduler that will cause the wrong order of task.
1. use different IRQ to let Ppl copy run in higher IRQ level.
2. clear scheduler IRQ earlier to let DAI IRQ schedule can run in time.
3. clear arch task IRQ earlier to make sure all task IRQ goes on time.

### CODE REVIEW

File: src/lib/schedule.c [hunk 1]
```diff
@@ -187,14 +187,14 @@ static struct task *schedule_edf(void)
 
 	tracev_pipe("edf");
 
+	interrupt_clear(PLATFORM_SCHEDULE_IRQ);
```
Comment (lgirdwood) [inline src/lib/schedule.c]: Can you check master for a similar patch by @tlauda and backport.

Comment (xiulipan) [reply src/lib/schedule.c]: I did not see any similar patch.
@tlauda 
Do you make some change like this in master?

Comment (lgirdwood) [reply src/lib/schedule.c]: Can you change the commit message, the real fix here is to move irq_clear before edf_get_next as there is a race when adding a new task.

**Comment by xiulipan** [comment]
@lgirdwood 
PR updated.
Test with
`aplay -D hw:0,0 -f dat /dev/zero & arecord -D hw:0,0 -f dat /dev/null -vv -i` and pressing space
No firmware xrun now.

**Pull request merged**